dll/win32/shell32/shlexec.cpp

```diff
@@ -782,17 +784,56 @@ static UINT SHELL_FindExecutable(LPCWSTR lpPath, LPCWSTR lpFile, LPCWSTR lpVerb,
         return 33;
     }
 
-    if (SearchPathW(lpPath, lpFile, L".exe", ARRAY_SIZE(xlpFile), xlpFile, NULL))
+    GetCurrentDirectoryW(ARRAY_SIZE(curdir), curdir);
+    if (!PathIsFileSpecW(lpFile))
     {
-        TRACE("SearchPathW returned non-zero\n");
-        lpFile = xlpFile;
-        /* The file was found in the application-supplied default directory (or the system search path) */
+        BOOL found = FALSE;
+        if (lpPath && *lpPath)
+        {
+            TRACE("ASDF %s\n", debugstr_w(lpPath));
+            PathCombineW(xlpFile, lpPath, lpFile);
+            if (PathFileExistsDefExtW(xlpFile, 0xbf) || PathFileExistsW(xlpFile))
```

Comment:
shlwapi_undoc.h has WHICH_ flags defined for these, don't need the magic 0xbf?

Reply:
Yep, we can use the proper flags here instead of this, but I took this code from Wine and since it's synced, it shouldn't have any difference(s) and shouldn't be modified. But if it can, it's not a problem to use the proper flags definitions here.

Reply:
This file is not wine synced (nothing in shell32 really is, just cherry pick things from upstream)

Reply:
OK then, added a proper flags for this call and the similar one below.